### DISCUSSION TITLE
Injections for templating languages

### DISCUSSION THREAD

**Discussion by rockerBOO** [nvim-treesitter/nvim-treesitter#3245] [Q&A] [answered]
The scenario I want to setup injections for templating languages like say embedding into HTML, and say with `eex`

```html
<div>
<%= "hello" %>
</div>
```

What would be the suggested approach? The `eex` parser doesn't know about HTML and HTML parser errors on this code. 

`set filetype=html`
![2022-08-01_06-09](https://user-images.githubusercontent.com/15027/182083165-608bf0ee-2887-49ca-962e-b74cc5e91c86.png)

`set filetype=eex`
![2022-08-01_06-10](https://user-images.githubusercontent.com/15027/182083287-4bd26939-b834-40be-81d7-de2db7e7273e.png)

Should I try to inject HTML surrounding the eex? Or somehow get the HTML parser to work better for these other templates? Is there another filetype interaction that might be similar to this?

Thank you!

**Comment by kyazdani42** [comment, marked as answer]
Thing is the html parser will fail for nodes starting with `<` since it tries to parse a tag that doesn't exist. I think a good approach would be to use the html parser as a base and extend it to parse `<%=` and its content.

**Reply by rockerBOO** [reply]
Thank you @kyazdani42 ! My problem with the HTML parser is I do not know how to properly query inside the "ERROR" section. Looking at all the other injections, I do not see how they are dealing with this. Would I need to fix the HTML parser to not error, and just leave an unmarked section?

Thank you!

**Reply by kyazdani42** [reply]
i don't how to you could override error nodes :smile: basically you should override the html parser by forking it, and matching tags that refer to eex so they don't give errors but nodes that you want.

**Reply by rockerBOO** [reply]
Thank you! I made a fork [tree-sitter-html-eex](https://github.com/rockerBOO/tree-sitter-html-eex/)
